app/routes/_index.tsx

```diff
@@ -1,15 +1,17 @@
-import { json } from "@remix-run/node";
+import { unstable_defineLoader } from "@remix-run/node";
 import type { MetaFunction } from "@remix-run/react";
 import { useLoaderData } from "@remix-run/react";
 
 import { EventDetails } from "~/components/EventDetails";
 import { PageGrid } from "~/components/PageGrid";
 import { constructSiteTitle } from "~/utils/common";
 
-export const loader = async () => {
+import eventsJson from "../data/events.json";
```

Comment:
This should get treeshaken out of the client bundle since it's only used in the `loader` - no need to do the async import which has issues on node without import assertions it seems

Reply:
Just out of curiosity, is import assertions going to eventually be the right thing to use?

Reply:
I think so?  IIRC I had issues using them currently though